### ISSUE TITLE
Problem with NFD normalized Unicode

### ISSUE THREAD

**Issue by tebeka** [prawnpdf/prawn#1154]
I'm writing a book on programming and want to show NFC & NFD normalized Unicode. The NFD version is not rendered well on PDF.

Here's a short example:
```
I once went to Kraków, it's a nice city. (NFC)

I once went to Kraków, it's a nice city. (NFD)
```

Here's the output (note the second line)
![adoc-nfd](https://user-images.githubusercontent.com/87697/77531226-42ec7380-6e9b-11ea-9cc6-0335260e0b4a.png)

Original discussion [here](https://discuss.asciidoctor.org/Problem-with-NFD-normalized-Unicode-in-PDF-td7780.html)

**Comment by gettalong** [comment]
Tl;dr: This is to be expected due to the current implementation of ttfunk.

NFD represents composed characters with multiple Unicode code points. This means that the application doing the rendering needs to understand that. Since PDF is, so to say, pre-rendered by the application creating the PDF, this work falls to Prawn and ttfunk.

Normally there is some information in the font file that allows an application to correctly place multiple glyphs representing a single character correctly. So to render such a character the application has to make use of this information which ttfunk currently doesn't. However, there is currently work going on to support more of the OpenType standard and then this *might* work.

**Comment by gettalong** [comment]
@tebeka Also see https://gist.github.com/gettalong/d894c29b551917573c708cd5110653a0

**Comment by tebeka** [comment]
Thanks!

**Comment by tebeka** [comment]
@gettalong I've tried to use `LibertinusMono` as the code font (need monospace font for code samples). It shows the umlaut but not on top of the `o`, see below. Any idea how to fix this?

![city-1](https://user-images.githubusercontent.com/87697/77617322-86e38500-6f3c-11ea-920a-e4237c3df8b2.png)

**Comment by tebeka** [comment]
FWIW: IBM Plex Mono (almost) does the job

![city-3](https://user-images.githubusercontent.com/87697/77656990-5243ed80-6f7d-11ea-86a3-789c76a6a769.png)

**Comment by gettalong** [comment]
@tebeka As mentioned in the link I posted, the behaviour depends entirely on the font. Each glyph defines a horizontal offset. For LibertinusMono that seems to be the "standard" offset whereas Plex Mono defines a negative offset so that the apostrophe overlaps the previous character.

In both cases no additional information from the font is used by Prawn/ttfunk. So this will probably never work with current versions of Prawn/ttfunk since the horizontal offset of the apostrophe has to vary based on the previous character. Such information is available in additional data structures inside the font but the library needs to understand that.

**Comment by pointlessone** [comment]
Prawn implement a very basic text layout algorithm. It basically takes individual glyphs and draws them one by one. This is why decomposed graphemes are rendered the way they are.

I'm closing this in favour of #1295 as it's basically what would take to resolve this.
